### PR TITLE
Support slack subscribe with conclusion filter #16

### PR DESCRIPTION
**Pull request by josephchow102**
connect #16

### CODE REVIEW

File: pkg/utils/array/unique.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	keys := make(map[T]bool)
+	list := []T{}
+	for _, entry := range strSlice {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
+			list = append(list, entry)
+		}
+	}
+	return list
+}
```
Comment (kiootic) [inline pkg/utils/array/unique.go:13]: Some Go idioms for your reference:
- it's common to use `map[T]struct{}` to represent sets
- usually zero values are use for initial empty slices (`var list []T`)
- the boolean flag of map index is usually named `ok`/`exists`

File: pkg/slack/app.go [hunk 2]
```diff
@@ -41,52 +48,93 @@ func (a *App) Disabled() bool {
...
 	} else if data == "" {
 		return nil, nil
 	}
-	return strings.Split(data, ";"), nil
+	channelInfoStrings := strings.Split(data, ";")
+	channelInfos := []ChannelInfo{}
+
+	for _, channelString := range channelInfoStrings {
+		channelID, conclusionsString, _ := strings.Cut(channelString, ":")
+		consclusions := []string{}
```
Comment (kiootic) [inline pkg/slack/app.go:72]: Typo

File: pkg/slack/app.go [hunk 3]
```diff
@@ -41,52 +48,93 @@ func (a *App) Disabled() bool {
...
+		}
+	}
+
+	if len(unsupportedConclusions) > 0 {
+		return fmt.Errorf("unsupported conclusions: %s", strings.Join(unsupportedConclusions, ", "))
+	}
+
+	newChannelInfoStrings := []string{}
+	for _, c := range channelInfos {
+		if c.channelID == channelInfo.channelID {
```
Comment (kiootic) [inline pkg/slack/app.go:108]: Should probably change duplication check into replacing it with with new conclusions, to support updating existing subscription directly?

Comment (josephchow102) [reply pkg/slack/app.go:108]: > Should probably change duplication check into replacing it with with new conclusions, to support updating existing subscription directly?

@kiootic Yes, I think this makes sense

cc @limouren

**Review by kiootic — APPROVED** [review]

**Comment by limouren** [comment]
@kiootic No concerns if I merge it?

**Comment by kiootic** [comment]
Sure no problem

**Pull request merged**